client/src/components/Ticketing/ticketingmanager/Events/showingInputContainer.tsx

```diff
@@ -7,36 +7,53 @@ import {Showing} from '../../../../interfaces/showing.interface';
  * Used to map props to input container correctly
  *
  * @module
- * @param {initialData} Showing
+ * @param {initialData} object
  * @param {number} id
- * @param {Function} addShow
- * @param {Function} deleteShow
+ * @param {Function} handleSetShow
  */
+interface InitialData {
+  availableseats?: number,
+  defaulttickettype?: number,
+  eventdate?: number | string,
+  eventid_fk?: number,
+  eventinstanceid?: number,
+  eventtime?: string,
+  ispreview?: boolean,
+  purchaseuri?: string,
+  salestatus?: boolean,
+  totalseats?: number
+}
+
 export interface MapPropsToShowingInputContainer {
-  initialData?: Showing;
+  initialData: InitialData;
   id: number;
   showingNum: number;
-  addShow: (show:Showing) => void;
-  deleteShow: (event: Event) => void;
+  handleSetShow: (show: Showing) => void;
 }
 
+
 /**
  *
- * @param {MapPropsToShowingInputContainer} {initialData, id, addShow, deleteShow}
+ * @param {MapPropsToShowingInputContainer} {initialData, id, handleSetShow}
  * @returns {ReactElement}
  */
 // eslint-disable-next-line react/prop-types
-const ShowingInputContainer = ({initialData, id, showingNum, addShow, deleteShow}:MapPropsToShowingInputContainer) => {
-  const [starttime, setStarttime] = useState(initialData.starttime !== undefined? initialData.starttime: '');
-  const [eventdate, setEventdate] = useState(initialData.eventdate !== undefined? initialData.eventdate: '');
-  const [ticketTypeId, setTicketTypeId] = useState('');
-  const [totalseats, setTotalseats] = useState(initialData.totalseats !== undefined? initialData.totalseats: 0);
-  const availableseates = initialData.availableseats !== undefined? initialData.availableseats: 0;
+
+const toDateStringFormat = (date) => {
```

Comment:
> @samkhodak we should use a Date object instead of a string for the date and change the type in the database as well in a separate PR. Can you make an issue for it?

I've made this into an issue, #179.